README.md

```diff
@@ -47,17 +47,52 @@ This proposal is a collection of multiple packages.
 
 ### Goals [or Motivating Use Cases, or Scenarios]
 
-[What is the end-user need which this project aims to address?]
+- Bring some of the benefits that Wasm provides to CPU code, over to GPU code. 
+- Enable secure GPU compute.
+- Enable wasm-based UI applications.
+- Enable wasm-based graphics even in environments with no GPUs.
+- Enable wasm-based AI for cases not covered by wasi-nn.
 
 ### Non-goals
 
-[If there are "adjacent" goals which may appear to be in scope but aren't, enumerate them here. This section may be fleshed out as your design progresses and you encounter necessary technical and other trade-offs.]
+- A full windowing API.
+- VR/AR (subject to change, if we find contributors with expertise in the field).
 
 ### API walk-through
 
 The full API documentation can be found in [imports.md](imports.md).
 
-[Walk through of how someone would use this API.]
+#### `wasi:webgpu`
+
+`wasi:webgpu` is based on the official [webgpu spec](https://www.w3.org/TR/webgpu/).
+
+`wasi:webgpu` does deviate sometimes from the webgpu spec, namely, in cases where the spec makes assumptions about running in a web or JS environment. Wherever `wasi:webgpu` deviates from the spec, a clear explanation should be documented.
```

Comment:
I agree we should say it deviates, but we should be careful to clarify that it varies only in syntax and not semantics. The goal is to remain 100% compatible with the functionality of WebGPU, with just minor syntactical differences from not having a JS/Web env.

Reply:
Fair point, feel free to edit the text